structs.inc

```diff
@@ -331,8 +331,32 @@ MACRO dstruct ; struct_type, instance_name[, ...]
 			def sizeof_\2    EQU @ - (\2)
 			structs_assert sizeof_\1 == sizeof_\2
 
-			IF def(STRUCTS_EXPORT_CONSTANTS)
-				EXPORT \2_nb_fields, sizeof_\2
+			IF def(STRUCT_SIZEOF_SPEC)
+				def STRUCT_SIZEOF_NAME EQUS STRFMT("{STRUCT_SIZEOF_SPEC}", "\2")
+				; Check in case STRUCT_SIZEOF_SPEC is "sizeof_%s"
+				IF !def({STRUCT_SIZEOF_NAME})
+					def {STRUCT_SIZEOF_NAME} EQU sizeof_\2
+				ENDC
+				IF def(STRUCTS_EXPORT_CONSTANTS)
+					EXPORT {STRUCT_SIZEOF_NAME}
+				ENDC
+				PURGE STRUCT_SIZEOF_NAME
+			ELIF def(STRUCTS_EXPORT_CONSTANTS)
+				EXPORT sizeof_\2
+			ENDC
+
+			IF def(STRUCT_NB_FIELDS_SPEC)
+				def STRUCT_NB_FIELDS_NAME EQUS STRFMT("{STRUCT_NB_FIELDS_SPEC}", "\2")
+				; Check in case STRUCT_NB_FIELDS_SPEC is "%s_nb_fields"
+				IF !def({STRUCT_NB_FIELDS_NAME})
+					def {STRUCT_NB_FIELDS_NAME} EQU \2_nb_fields
+				ENDC
+				IF def(STRUCTS_EXPORT_CONSTANTS)
+					EXPORT {STRUCT_NB_FIELDS_NAME}
+				ENDC
+				PURGE STRUCT_NB_FIELDS_NAME
+			ELIF def(STRUCTS_EXPORT_CONSTANTS)
+				EXPORT \2_nb_fields
```

Comment:
Couldn't the same be achieved with a wrapper macro?
```
MACRO my_dstruct
    dstruct \#
    export def \2_SIZE equ sizeof_\2
    export def \2_COUNT equ \2_nb_fields
ENDM
```

That'd avoid adding complexity to this macro.

Reply:
Sure, I'll revert that commit.